evm/executeContract.js

```diff
@@ -0,0 +1,244 @@
+        throw new Error('Missing token value from user info');
+    }
+
+    const IContractExecutor = require(callContractPath);
+    const contract = new Contract(callContractAddress, IContractExecutor.abi, wallet);
+    var callData, tokenValue;
+
+    if (methodName === 'default') {
+        callData = inputCallData;
+        tokenValue = inputNativeValue;
+    } else {
+        const ITargetContract = require(targetContractPath);
+        const targetContract = new Contract(targetContractAddress, ITargetContract.abi, wallet);
+        callData = await getCallData(methodName, targetContract, recipient, amount);
+        tokenValue = Number(0);
+    }
+
+    (async () => {
+        try {
+            const result = await contract.executeContract(targetContractAddress, callData, tokenValue);
+            printLog('Function successfully called with return value as: ');
+            printObj(result);
+        } catch (error) {
+            printError('Calling executeContract method failed with Error: ', error);
+        }
+    })();
+}
+
+async function main(options) {
+    const config = loadConfig(options.env);
+    let chains = options.chainNames.split(',').map((str) => str.trim());
+
+    if (options.chainNames === 'all') {
+        chains = Object.keys(config.chains);
+    }
+
+    for (const chain of chains) {
+        if (config.chains[chain.toLowerCase()] === undefined) {
+            throw new Error(`Chain ${chain} is not defined in the info file`);
+        }
+    }
+
+    for (const chain of chains) {
+        const rpc = config.chains[chain.toLowerCase()].rpc;
+        const provider = getDefaultProvider(rpc);
+        const privateKey = options.privateKey;
+
+        if (!isString(privateKey)) {
+            throw new Error('Private Key value is not provided in the info file');
+        }
+
+        const wallet = new Wallet(privateKey, provider);
+        await executeContract(
+            options.pathCallContract,
+            options.pathTargetContract,
+            options.callContractAddress,
+            options.targetAddress,
+            wallet,
+            options.callData,
+            Number(options.nativeValue),
+            options.methodName,
+            options.recipientAddress,
+            Number(options.amount),
+        );
+        saveConfig(config, options.env);
+    }
+}
+
+const program = new Command();
+
+program.name('execute-contract').description('Executes a call to an external contract');
+
+program.addOption(
+    new Option('-e, --env <env>', 'environment')
+        .choices(['local', 'devnet', 'stagenet', 'testnet', 'mainnet'])
+        .default('testnet')
+        .makeOptionMandatory(true)
+        .env('ENV'),
+);
+program.addOption(new Option('-n, --chainNames <chainNames>', 'chain names').makeOptionMandatory(true));
+program.addOption(
+    new Option('-pc, --pathCallContract <pathCallContract>', 'artifact path for the called contract').makeOptionMandatory(true),
```

Comment:
Do we want to make this optional and add the Operators contract path as the default value? If we expect this script to be used for the Operators contract execute function in most cases

Reply:
@deanamiel as far as I understand artifact path of the called contract would come from a different repo, and its location can change based on the user's setup of repositories. Setting a default path can lead to dependencies between two repositories, which might not be favourable.

If my understanding is incorrect, please let me know. This was my first script for the repo.

Reply:
For now, I will be following this pattern from the `deploy-contract.js` script. But here as well there is no default artifact location. I can make the tag optional and set different artifact paths based on `contractName` option (once I add it) of the script, but for it I would like to understand local setup preferred by all devs working on these modules.